luaui/configs/unit_buildmenu_config.lua

```diff
@@ -3,22 +3,18 @@
 --- DateTime: 4/26/2023 8:48 PM
 ---
 
-
-local unitName = {}
-local unitEnergyCost = {}
-local unitMetalCost = {}
-local unitGroup = {}
-local unitRestricted = {}
-local isBuilder = {}
-local isFactory = {}
-local unitIconType = {}
-local isMex = {}
-local isWind = {}
-local isWaterUnit = {}
-local isGeothermal = {}
-local unitMaxWeaponRange = {}
-
-local showWaterUnits = false
```

Comment:
you removed showWaterUnits?

Reply:
It's only ever false and not read by grid menu or classic build menu. Seems like some forgotten cruft.

Reply:
In the case of this variable, it seems like there are two files importing the buildmenu_config (gridmenu and buildmenu, indeed)... they do use a variable very suspiciously similarly named showWaterUnits, just not from the import.

Imo probably both files could use the value from here in case someone wants to touch that for debugging or smth. Ie, doing local showWaterUnits = units.showWaterUnits in both files, and then doing a similar check for showWaterUnits before restrictingWaterUnits... for consistency XDD

No idea why someone would want to do that, also why the test exists at the top in just one of the files just after setting it.

The change you are doing is definitely safe, but it's difficult to say if it's in the good direction or not XD.

Reply:
Hm. The variable is always false, it can only be misleading.

https://github.com/beyond-all-reason/Beyond-All-Reason/commit/3ba56c83c30705ed8a294e05d6b26e75cfce7af5

I assume it was some half-written code or logic that was moved into the build menu scripts.

In my opinion removing dead/misleading code is a good thing. It might be worth moving any shared logic into this file, but it's out of scope for this PR.

Reply:
Yeah it's not a big deal imo, just saying maybe separate PR addressing that and other places where the variable is used weirdly (like where it [gets tested](https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/luaui/Widgets/gui_buildmenu.lua#L220) after set to false) might have been the way.

Removing dead or misleading code is indeed good, but doing it in PRs focusing in something different imo not really always good since that makes PRs more cumbersome to review. Specially when the case is not 100% clear cut like here (even if it's 95% clear cut I agree with you in that).

I mean, It's something we all do sometimes, but personally at least most of the time I try to refrain from touching other little things while I'm at it. When I feel like I'm changing already a lot I may overreach too, but even then I would not just touch random things mostly things I feel are somehow relevant to the PR.

In any case, I hope we both understand each other XD, I don't care much about the variable tbh XD, but on first review I felt like that could be an error since when adding annotations you're specially not supposed to change code. Also had to track all the reordering to make sure code wasn't changed.

Reply:
#3935